keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+        if survival_probability:
+            x = layers.Dropout(
+                survival_probability, noise_shape=(None, 1, 1, 1), name=name + "drop"
+            )(x)
+
+        x = layers.Add()([x, shortcut])
+
+        return layers.Activation(activation, name=name + "output_act")(x)
+
+    return apply
+
+
+def BlockGroup(
+    filters,
+    strides,
+    se_ratio,
+    bn_epsilon,
+    bn_momentum,
+    num_repeats,
+    activation,
+    survival_probability: float,
+    name: str,
+):
+    """Create one group of blocks for the ResNet model."""
+
+    def apply(inputs):
+        # Only the first block per block_group uses projection shortcut and strides.
+        x = BottleneckBlock(
+            filters=filters,
+            strides=strides,
+            use_projection=True,
+            se_ratio=se_ratio,
+            bn_epsilon=bn_epsilon,
+            bn_momentum=bn_momentum,
+            activation=activation,
+            survival_probability=survival_probability,
+            name=name + "block_0_",
+        )(inputs)
+
+        for i in range(1, num_repeats):
+            x = BottleneckBlock(
+                filters=filters,
+                strides=1,
+                use_projection=False,
+                se_ratio=se_ratio,
+                activation=activation,
+                bn_epsilon=bn_epsilon,
+                bn_momentum=bn_momentum,
+                survival_probability=survival_probability,
+                name=name + f"block_{i}_",
+            )(x)
+        return x
+
+    return apply
+
+def get_survival_probability(init_rate, block_num, total_blocks):
+    """Get survival probability based on block number and initial rate.
+    """
+    return init_rate * float(block_num) / total_blocks
+
+
+def allow_bigger_recursion(target_limit: int):
+    """Increase default recursion limit to create larger models."""
+    current_limit = sys.getrecursionlimit()
+    if current_limit < target_limit:
+        sys.setrecursionlimit(target_limit)
+
+
+def fixed_padding(inputs, kernel_size):
+    """Pad the input along the spatial dimensions independently of input size."""
+    pad_total = kernel_size - 1
+    pad_beg = pad_total // 2
+    pad_end = pad_total - pad_beg
+
+    # Use ZeroPadding as to avoid TFOpLambda layer
+    padded_inputs = layers.ZeroPadding2D(
+        padding=((pad_beg, pad_end), (pad_beg, pad_end))
+    )(inputs)
+
+    return padded_inputs
+
+def ResNetRS(
+    depth: int,
+    input_shape=None,
+    bn_momentum=0.0,
+    bn_epsilon=1e-5,
+    activation: str = "relu",
+    se_ratio=0.25,
+    dropout_rate=0.25,
+    drop_connect_rate=0.2,
+    include_top=True,
+    block_args: List[Dict[str, int]] = None,
+    model_name="resnet-rs",
+    pooling=None,
+    weights="imagenet",
+    input_tensor=None,
+    classes=1000,
+    classifier_activation: Union[str, Callable] = "softmax",
+    include_preprocessing=True
+):
+    """Build Resnet-RS model, given provided parameters.
+    Parameters:
```

Comment:
The docstring formatting should be fixed (see other docstrings in the Keras codebase for examples)

Reply:
(Not yet done, please fix)

Reply:
Another important note is that all these argument descriptions are redundant with the descriptions provided above. There should be a single source of truth so we don't have to copy/paste too much content when making modifications.